services/ui-src/src/components/overlays/EntityDetailsOverlay.tsx

```diff
@@ -1,176 +1,123 @@
-      <Box data-testid="entity-details-overlay">
-        <Box
-          as="button"
-          sx={sx.backButton}
-          onClick={closeOverlay}
-          aria-label="Return to MLR reporting"
-        >
-          <Image src={arrowLeftBlue} alt="Arrow left" sx={sx.backIcon} />
-          Return to MLR Reporting
-        </Box>
-        {verbiage.intro && (
-          <ReportPageIntro
-            text={overlayVerbiage.MLR.intro}
-            accordion={accordionVerbiage.MLR.detailIntro}
-            reportType={report?.reportType}
-          />
-        )}
-        <Box sx={sx.programInfo}>
-          <Text sx={sx.textHeading}>MLR report for:</Text>
-          <ul>
-            {programInfo.map((field, index) => (
-              <li key={index}>{field}</li>
-            ))}
-          </ul>
-        </Box>
-        <Form
-          id={form.id}
-          formJson={form}
-          onSubmit={onSubmit}
-          onError={onError}
-          formData={selectedEntity}
-          autosave={true}
-          disabled={!userIsEndUser}
-          validateOnRender={validateOnRender || false}
-          dontReset={false}
-        />
-        <Box sx={sx.footerBox}>
-          <Flex sx={sx.buttonFlex}>
+    <Box>
+      <Button
+        sx={sx.backButton}
+        variant="none"
+        onClick={closeEntityDetailsOverlay as MouseEventHandler}
+        aria-label="Return to MLR reporting"
+      >
+        <Image src={arrowLeftBlue} alt="Arrow left" sx={sx.backIcon} />
+        Return to MLR Reporting
+      </Button>
+      <ReportPageIntro
+        text={overlayVerbiage.MLR.intro}
+        accordion={accordionVerbiage.MLR.detailIntro}
+      />
+      <Box sx={sx.programInfo}>
+        <Text sx={sx.textHeading}>MLR report for:</Text>
+        <ul>
+          {programInfo.map((field, index) => (
+            <li key={index}>{field}</li>
+          ))}
+        </ul>
+      </Box>
+      <Form
+        id={form.id}
+        formJson={form}
+        onSubmit={onSubmit}
+        formData={selectedEntity}
+        autosave={true}
+        disabled={disabled}
+        validateOnRender={validateOnRender || false}
+        dontReset={true}
+      />
+      <Box sx={sx.footerBox}>
+        <Flex sx={sx.buttonFlex}>
+          {disabled ? (
             <Button
-              onClick={() => closeOverlay()}
-              type="submit"
-              sx={sx.saveButton}
+              variant="outline"
+              onClick={closeEntityDetailsOverlay as MouseEventHandler}
             >
-              {submitting ? (
-                <Spinner size="md" />
-              ) : report?.locked ? (
-                "Return"
-              ) : (
-                "Save & return"
-              )}
+              Return
+            </Button>
+          ) : (
+            <Button type="submit" form={form.id} sx={sx.saveButton}>
+              {submitting ? <Spinner size="md" /> : "Save & return"}
             </Button>
-          </Flex>
-        </Box>
+          )}
```

Comment:
I've cleaned up the button and simplified the states for it. It should now be inline with how our other buttons (See the drawer for an example) act when an admin is on the page